presets/ragengine/vector_store/base.py

```diff
@@ -381,20 +422,21 @@ async def chat_completion(self, request: dict) -> ChatCompletionResponse:
         logger.info(
             f"Creating chat engine for index '{request.get('index_name')}' with prompt size: {prompt_len}"
         )
-        chat_engine = self.index_map[
-            request.get("index_name")
-        ].as_chat_engine(
+        # Calculate top_k based on available context. For larger windows, we can afford to retrieve more documents.
+        # 750 tokens is a rough estimate based off max 1500 for code and 1000 for text
+        top_k = max(100, int((self.llm.metadata.context_window - prompt_len) / 750))
+        chat_engine = self.index_map[request.get("index_name")].as_chat_engine(
             llm=self.llm,
-            similarity_top_k=100,  # Might want to make this a function of avg doc node size in an index but this should be a wide enough default
-            chat_mode=ChatMode.CONDENSE_PLUS_CONTEXT,
+            similarity_top_k=top_k,
+            chat_mode=ChatMode.CONTEXT,
```

Comment:
What is the difference between the CONDENSE_PLUS_CONTEXT vs CONTEXT?

Reply:
CONDENSE_PLUS_CONTEXT condenses chat history and tries to run vector search off all chat + history. Context will only add the context on the latest chat message.